packages/wrangler/src/event-subscriptions/index.ts

```diff
@@ -0,0 +1,323 @@
+	metadata: {
+		owner: "Product: Queues",
+		status: "open-beta",
+		description: "Manage Event Subscriptions",
+	},
+});
+
+export const eventSubscriptionsBrowseCommand = createCommand({
+	metadata: {
+		owner: "Product: Queues",
+		status: "open-beta",
+		description: "List available services and events",
+	},
+
+	args: {
+		source: {
+			type: "string",
+			description: "Filter by service",
+			demandOption: false,
+		},
+	},
+
+	async handler(args, ctx) {
+		const config = readConfig(args);
+		const accountId = await requireAuth(config);
+
+		const events = await ctx.fetchResult<EventSubscriptionEventSpec[]>(
+			`/accounts/${accountId}/event_subscriptions/events`
+		);
+
+		let products = PRODUCTS.map((p) => ({
+			...p,
+			events: events.find((e) => e.id === p.id)?.events,
+		})).filter((p) => p.events && p.events.length !== 0);
+
+		if (args.source) {
+			const sources = args.source
+				.split(",")
+				.map((s) => s.toLocaleLowerCase().trim());
+			products = products.filter(
+				(p) =>
+					sources.includes(p.id.toLocaleLowerCase()) ||
+					sources.includes(p.name.toLocaleLowerCase())
+			);
+		}
+
+		if (products.length === 0) {
+			return logger.table([{ service: "", event: "", description: "" }]);
+		}
+
+		logger.table(
+			products.flatMap((p) =>
+				// Disabling ESLint and using `!` because we've validated that `events` is not null
+				// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+				p.events!.map((e) => ({
+					service: p.name,
+					event: e.name,
+					description: e.description,
+				}))
+			)
+		);
+	},
+});
+
+export const eventSubscriptionsListCommand = createCommand({
+	metadata: {
+		owner: "Product: Queues",
+		status: "open-beta",
+		description: "List configured Event Subscriptions",
+	},
+
+	async handler(args, ctx) {
+		const config = readConfig(args);
+		const accountId = await requireAuth(config);
+
+		const res = await ctx.fetchResult<EventSubscription[]>(
+			`/accounts/${accountId}/event_subscriptions/subscriptions?per_page=100`
+		);
+
+		if (res.length === 0) {
+			return logger.table([
+				{ id: "", name: "", source: "", destination: "", events: "" },
+			]);
+		}
+
+		logger.table(
+			res.map((s) => ({
+				id: s.id,
+				name: s.name,
+				source: `${s.product_id}.${s.resource_id}`,
+				destination: `queues.${s.queue_name}`,
+				events: s.events.join(", "),
+			}))
+		);
+	},
+});
+
+export const eventSubscriptionsGetCommand = createCommand({
+	metadata: {
+		owner: "Product: Queues",
+		status: "open-beta",
+		description: "Fetch details about a single Event Subscription",
+	},
+
+	positionalArgs: ["id"],
+	args: {
+		id: {
+			type: "string",
+			description: "ID of the Event Subscription",
+			demandOption: true,
+		},
+	},
+
+	async handler(args, ctx) {
+		const config = readConfig(args);
+		const accountId = await requireAuth(config);
+		const { id } = args;
+
+		const subscription = await ctx.fetchResult<EventSubscription>(
```

Comment:
what does the API return if no subscription with `id` is found? should that case be handled by the client here?

Reply:
The API would return a 404 with an error message. I looked up what the `fetchResult` function was doing, and it looks like it handles that already?
But I think I'll  add a test for this case, why not